lib/cfg.js

```diff
@@ -0,0 +1,368 @@
+    var actPosSz = actionPos.length;
+    if (actionPos[actPosSz-1] + 1 == basicBlock.end){
+        --actPosSz;
+    }
+
+    if(actPosSz === 0)
+        return [{
+                    nameId: basicBlock.nameId,
+                    start: basicBlock.start,
+                    end: basicBlock.end
+               }];
+    else if(actPosSz == 1)
+            return [
+                {nameId: basicBlock.nameId,start: basicBlock.start,end:actionPos[0]+1},
+                {nameId: basicBlock.nameId+"@"+ (actionPos[0]+1),start: actionPos[0]+1,end: basicBlock.end}
+               ];
+    else {
+        var first = 0;
+        var last = actPosSz;
+        var block_name = basicBlock.nameId;
+        var tmp = {nameId:block_name, start:basicBlock.start, end:actionPos[first]+1 };
+        var result = [];
+        result.push(JSON.parse(JSON.stringify(tmp)));
+        while(first != last-1) {
+            tmp.nameId = block_name + "@" + (actionPos[first]+1);
+            tmp.start = actionPos[first]+1;
+            ++first;
+            tmp.end = actionPos[first]+1;
+            result.push(JSON.parse(JSON.stringify(tmp)));
+        }
+        
+        tmp = {nameId:block_name + "@" + (actionPos[first] + 1), start:actionPos[first] + 1, end:basicBlock.end };
+        result.push(JSON.parse(JSON.stringify(tmp)));
+        
+        return result;
+
+    }
+
+};
+
+ControlFlowGraph.prototype.concatInstructions = function (asmArr, first, last) {
+    
+    var makeShorter = function(instruction){
+        if(instruction.length > 50){
+            return instruction.substr(0, 50);
+        }
+        
+        return instruction;
+    };
+    
+    if(first == last) return "";//if last -1 is changed to last this line is no longuer needed
+    console.log("look");
+    var result = "";
+    //TODO: include \n for long templated instructions
+    while(first != last-1) {//added to delete last \n and handle the last concat outside loop
+        console.log(asmArr[first].text);
+        result += makeShorter(asmArr[first].text) + "\n";
+        ++first;
+    }
+    //last concat withou \n
+    result += makeShorter(asmArr[first].text);
+    console.log(asmArr[first].text);
+
+    return result;
+};
+
+ControlFlowGraph.prototype.makeNodes = function (asmArr, arrOfCanonicalBasicBlock) {
+    var node = {};
+    var nodes = [];
+    
+    _.each(arrOfCanonicalBasicBlock, _.bind(function(x){
+        console.log("node name:");
+        console.log(x.nameId);
+        node.id = x.nameId;
+        node.label = x.nameId + ((x.nameId.indexOf(":") !== -1)? "":":") +"\n"+
+                         this.concatInstructions(asmArr,x.start, x.end);
+        node.color = "#99ccff";//"#FFAFAF";
+        node.shape = 'box';
+        nodes.push(JSON.parse(JSON.stringify(node)));
+    }, this));
+    return nodes;
+};
+
+ControlFlowGraph.prototype.makeEdges = function (asmArr, arrOfCanonicalBasicBlock, instructionType, extractNodeNameFromInstruction) {
+
+    var jmpInst = 0;
```

Comment:
Ah, as described above, move these to global scope and use them in all places instead of the magic numbers. Thanks!

Reply:
done